scripts/update_production.sh

```diff
@@ -6,31 +6,54 @@
 SCRIPT_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 PROJECT_ROOT=$(realpath ${SCRIPT_PATH}/..)
 
+# Keep track of any errors that occur during the update.
+STATUS=0
+trap 'STATUS=$?' ERR
+
 # Virtuoso has a checkpoint interval of 60 minutes. That means that every 60 minutes,
 # the server will be un responsive for a small amount of time (a few seconds).
 # Since our updates take too long, this can lead to "random" valid query failures.
 # Set the interval to 0 to disable automatic checkpoints. At the end of the update
 # The interval will be restored, and a manual checkpoint will be fired in order to
 # write all the transactions of the updates into the disk.
-echo "Disable automatic checkpoints."
+echo "Disabling automatic checkpoints."
 ./vendor/bin/phing set-virtuoso-checkpoint -Dinterval="-1"
 
 # Perform the necessary steps for the update
 cd ${PROJECT_ROOT}
+
+# Make sure config is writable when performing updates. This depends on the
+# following code being present in web/sites/default/settings.php:
+# $settings['config_readonly'] = !file_exists(getcwd() . '/../disable-config-readonly');
+grep -Fqx '$settings['\''config_readonly'\''] = !file_exists(getcwd() . '\''/../disable-config-readonly'\'');' web/sites/default/settings.php
+
+if [ ${STATUS} -ne 0 ]; then
+  echo "The following line is missing from web/sites/default/settings.php:"
+  echo '$settings['\''config_readonly'\''] = !file_exists(getcwd() . '\''/../disable-config-readonly'\'');'
+  exit ${STATUS}
+fi
+
+echo "Disabling config_readonly."
+touch disable-config-readonly
+
 ./vendor/bin/drush updatedb --yes &&
```

Comment:
Well, I have a problem with this approach. I find it better to exit with an error after each command failure. If `updatedb` is failing, for instance, I don't want to continue.  It just happened to me 2 days ago -- a build failed in `updb` but it took me 10 minutes to read the console log starting from bottom, till I found the cause. But probably this out-of-scope? I would like to hear your opinion.

Reply:
Yes this has been restored in this PR. All commands that are time intensive are chained together again, and I have introduced an error trap that will cause any error to cause the build to fail.